isis/tests/AngleTest.cpp

```diff
@@ -0,0 +1,304 @@
+  EXPECT_FALSE(angle.isValid());
+  EXPECT_STREQ(angle.toString().toLatin1(), ""); 
+}
+
+
+TEST(AngleTest, DegreeInput_RadianOutput) {
+  Isis::Angle angle(30., Isis::Angle::Degrees );
+  EXPECT_NEAR(angle.radians(), 30 * Isis::PI/180.0, TEST_EPSILON);
+  EXPECT_TRUE(angle.isValid()); 
+  EXPECT_STREQ(angle.toString().toLatin1(), "30.0 degrees");  
+}
+
+
+TEST(AngleTest, RadianInput_DegreeOutput) {
+  Isis::Angle angle(30. * Isis::PI / 180.0, Isis::Angle::Radians );
+  EXPECT_NEAR(angle.degrees(), 30.0, TEST_EPSILON);
+  EXPECT_STREQ(angle.toString(false).toLatin1(), "30.0");  
+}
+
+
+TEST(AngleTest, CopyConstructor) {
+  Isis::Angle angle(30., Isis::Angle::Degrees );
+  Isis::Angle angleCopy(angle);
+  EXPECT_NEAR(angleCopy.degrees(), 30.0,TEST_EPSILON); 
+}
+
+
+TEST(AngleTest, Mutators) {
+  Isis::Angle angle(30.0, Isis::Angle::Degrees );
+  angle.setDegrees(180);
+  EXPECT_NEAR(angle.degrees(), 180.0, TEST_EPSILON);
+  angle.setRadians(Isis::PI);
+  EXPECT_NEAR(angle.radians(), Isis::PI, TEST_EPSILON); 
+}
+
+TEST(AngleOperators, QDebug) {
+  // Stream qDebug output to a string and then do a string comparison?
+  Isis::Angle angle(45.0, Isis::Angle::Degrees);
+  QString qdebug_output;
+  QDebug debug(&qdebug_output);
+  debug << angle;
+  EXPECT_STREQ(qdebug_output.toLatin1(), "0.785398 <radians> (45 <degrees>) "); 
+}
+
+
+TEST(AngleOperators, Assignment) {
+  Isis::Angle angle(30.0, Isis::Angle::Degrees );
+  angle = Isis::Angle(45.0, Isis::Angle::Degrees);
+  EXPECT_NEAR(angle.degrees(), 45.0, TEST_EPSILON); 
+}
+
+
+TEST(AngleOperators, AddSubtract) {
+  Isis::Angle angle1(30.0, Isis::Angle::Degrees );
+  Isis::Angle angle2(60.0, Isis::Angle::Degrees );
+  angle1 = angle1 + angle2;  
+  EXPECT_NEAR(angle1.degrees(), 90.0, TEST_EPSILON);
+  angle1 += angle2;
+  EXPECT_NEAR(angle1.degrees(), 150.0, TEST_EPSILON);
+  angle1 -= angle2;
+  EXPECT_NEAR(angle1.degrees(), 90.0, TEST_EPSILON);    
+  angle1 = angle1 - angle2;
+  EXPECT_NEAR(angle1.degrees(), 30.0, TEST_EPSILON);
+}
+
+
+TEST(AngleOperators, MultiplyDivide) {
+  Isis::Angle angle(30., Isis::Angle::Degrees );
+  angle = 2. * angle;
+  EXPECT_NEAR(angle.degrees(), 60.0, TEST_EPSILON);
+  angle *= 2;
+  EXPECT_NEAR(angle.degrees(), 120.0, TEST_EPSILON);
+  angle /= 2;
+  EXPECT_NEAR(angle.degrees(), 60.0, TEST_EPSILON);    
+  angle = angle / 2;
+  EXPECT_NEAR(angle.degrees(), 30.0, TEST_EPSILON);
+}  
+
+
+TEST(AngleOperators, Logical) {
+  Isis::Angle angle1(30.0, Isis::Angle::Degrees );
+  Isis::Angle angle2(45.0, Isis::Angle::Degrees );
+  EXPECT_FALSE(angle1 == angle2);
+  EXPECT_TRUE(Isis::Angle() == Isis::Angle()); 
+  EXPECT_FALSE(Isis::Angle() == angle2); 
+}
+
+
+TEST(AngleOperators, LessThan) {
+  Isis::Angle angle1(30.0, Isis::Angle::Degrees );
+  Isis::Angle angle2(45.0, Isis::Angle::Degrees );
+  EXPECT_TRUE(angle1 < angle2);
+  EXPECT_FALSE(angle2 < angle1);
+  Isis::Angle epsilon(1.1920929e-10, Isis::Angle::Degrees);
+  EXPECT_TRUE(angle1 < angle1 + epsilon); 
```

Comment:
I don't think this epsilon part is required. The third case would be angle1 < angle1 == false

Reply:
Similar to the LessThanOrEqualTo test

Reply:
I did get rid of this test in the updated version, but I think I figured out why this was in there in the first place. The original test actually did something like: 

```
  EXPECT_FALSE(angle1 < angle1 + epsilon); 
```

So perhaps the original test was testing something like: "this epilson is so small that when added to `angle1`, it doesn't increase its value to measurable more than `angle1`... I'm not sure why were were testing this, but it looks like that was what we were doing.

Reply:
I really don't think that's worth testing, unless we are doing some weird internal tolerances in the comparison.

Reply:
Glad to hear it, as I did just get rid of this.